app/graphql/mutations/users/register_user.rb

```diff
@@ -0,0 +1,37 @@
+module Mutations
+  module Users
+    class RegisterUser < BaseMutation
+      class AuthProviderSignupData < Types::BaseInputObject
+        argument :credentials, Types::AuthProviderCredentialsInput, required: true
+      end
+
+      argument :first_name, String, required: true
+      argument :last_name, String, required: true
+      argument :street, String, required: true
+      argument :city, String, required: true
+      argument :state, String, required: true
+      argument :zip, String, required: true
+      argument :image, String, required: true
+
+      argument :auth_provider, AuthProviderSignupData, required: true
+
+      # type Types::UserType
+      field :user, Types::UserType, null: false
+
+      def resolve(auth_provider: nil, first_name: nil, last_name: nil, street: nil, city: nil, state: nil, zip: nil, image: nil)
+        user = User.create!(
+          username: auth_provider&.[](:credentials)&.[](:username),
+          password: auth_provider&.[](:credentials)&.[](:password),
```

Comment:
This `&.` syntax may actually help me debug the nil error on the user update mutation that I'm currently working on. Being able to use this operator as a nil check is a nice tool

Reply:
This is interesting, so it's basically just checking that username and password are not nil?

Reply:
Trust... I didn't come up with that lol. But yes that's my understanding!